tests/test_process_response.py

```diff
@@ -0,0 +1,18 @@
+from typing_extensions import TypedDict
+from pydantic import BaseModel
+from instructor.process_response import handle_response_model
+
+
+def test_typed_dict_conversion() -> None:
+    class User(TypedDict):  # type: ignore
+        name: str
+        age: int
+
+    _, user_tool_definition = handle_response_model(User)
+
+    class User(BaseModel):
+        name: str
+        age: int
```

Comment:
Does it work with Required/NotRequired types? 🤔

Reply:
Hmm currently with my approach, I assume all keys are required but I see where you're coming from. Will try to see what I can do

Reply:
is optional not supported?